### ISSUE TITLE
Confidence score of OCR

### ISSUE THREAD

**Issue by ravikantb** [bgshih/crnn#26]
Hi,

Is there any way we can get some kind of confidence score for the OCR being done by CRNN?
If it is not there then can someone suggest a possible way to do it?

Thanks

**Comment by oneOfThePeople** [comment]
when you get output from forward image in the model ,like this:

`local output = model:forward(image)`

you need to get the max value of each possible position ,like this:
```
y, i = torch.max(output, 3)
```
i = index => the label (class)
y = values => the "confidens" when the max value it can be is 0

for check you can run demo.lua and change recognizeImageLexiconFree function in utilities.lua:
```
function recognizeImageLexiconFree(model, image)
    --[[ Lexicon-free text recognition.
    ARGS:
      - `model`   : CRNN model
      - `image`   : single-channel image, byte tensor
    RETURN:
      - `str`     : recognized string
      - `rawStr`  : raw recognized string
    ]]
    assert(image:dim() == 2 and image:type() == 'torch.ByteTensor','Input image should be single-channel byte tensor')
    image = image:view(1, 1, image:size(1), image:size(2))
    local output = model:forward(image)
    y, i = torch.max(output, 3)
    print ('------label------')
    print (i)
    print ('------conf------')
    print (y)
    local pred, predRaw = naiveDecoding(output) 
    print ('------pred------')
    print (pred)
    local str = label2str(pred)[1]
    local rawStr = label2str(predRaw, true)[1]
    return str, rawStr
end

```

**Comment by ravikantb** [comment]
@oneOfThePeople Thanks a lot! It helped :)

**Comment by ahmedmazari-dhatim** [comment]
Hi @oneOfThePeople @ravikantb  ,
Thank you for this clarification. 
 1)l would like to understand the meaning of value `3` here

`y, i = torch.max(output, 3)`

2) By ` 'i' `the label index you mean the label of all the sequence or label for each character of a sequence ? 

3) the value y returned by  torch.max(output, 3) is a logit since max value is 0. So how can l get the probability instead of logits ?

**Comment by vinayakarannil** [comment]
But this approach doesn't give the confidence score for the word as a whole right?
